src/lib.rs

```diff
@@ -2361,6 +2358,12 @@ impl Url {
 
     // Private helper methods:
 
+    fn cannot_have_username_password_port(&self) -> bool {
+        self.host().unwrap_or(Host::Domain("")) == Host::Domain("")
```

Comment:
It would be nice to link https://url.spec.whatwg.org/#url-miscellaneous in the docstring for this method.